rust/src/treefile.rs

```diff
@@ -1471,8 +1458,9 @@ impl Treefile {
     pub(crate) fn write_compose_json(&self, rootfs_dfd: &openat::Dir) -> Result<()> {
         let target = Path::new(COMPOSE_JSON_PATH);
         rootfs_dfd.ensure_dir_all(target.parent().unwrap(), 0o755)?;
-        rootfs_dfd.write_file_contents(target, 0o644, self.serialized.as_bytes())?;
-        Ok(())
+        rootfs_dfd.write_file_with_sync(target, 0o644, |w| {
```

Comment:
Not a big deal, just noting I don't think we need to or should `fsync` here.
We'll clean this up later by using `atomic_write` anyways.

Reply:
Yeah, good point. I think I went with the `_sync` variant because I thought `write_file_contents` also synced, but it clearly doesn't.